nixos/tests/matrix/mjolnir.nix

```diff
@@ -109,8 +109,8 @@ import ../make-test-python.nix (
         environment.systemPackages = [
           (pkgs.writers.writePython3Bin "create_management_room_and_invite_mjolnir"
             { libraries = with pkgs.python3Packages; [
-                matrix-nio
-              ] ++ matrix-nio.optional-dependencies.e2e;
+                (matrix-nio.override { withOlm = true; })
+              ];
```

Comment:
Can't we use the old approach here of adding optional dependencies? That we we'd no longer risk having two distinct matrix-nios / mautrix packages in an env.

Reply:
That would mean that the libolm functionality would be used without its tests having been run, which is the invariant I’m trying to maintain. We could just disable the encryption‐related tests unconditionally, but given that it is probably going to be quite a long time before all of these bridges are using something else, it’s pretty critical functionality, and there are a lot of those tests, that feels like it would be a regression. I’m happy to do that if people really don’t like this solution and don’t think testing the encryption functionality is important, and I’m up for considering any better way to achieve the invariant, but that’s why it’s structured the way it is currently.

Reply:
Apologies, I missed that part above!

I agree that the additional safety-net of running the tests then is a good idea. And Sandro is right, the setup hooks should catch conflicting packages in the first place. While I still think the solution is kinda messy, I don't know a better way to do this and the situation we're in is also kind of messy.

One may argue that we could just keep these two packages being marked as insecure, but apparently most of the other folks here have a different take on this and it's not a hill worth dying on.

Resolving.